### DISCUSSION TITLE
Combo box width

### DISCUSSION THREAD

**Discussion by GoldsteinE** [emilk/egui#177] [Q&A]
I have a `egui::combo_box()` with variants (created via `ui.selectable_value()`) having different widths. Combo box seems to have width of the current selected option. I would like it to always have width of the widest option. Is this possible?

**Comment by Wumpf** [comment]
Running into this as well and was about to open a bug, but maybe I'm missing something here. To illustrate the issue:

short string is selected, expanding combobox
![image](https://user-images.githubusercontent.com/1220815/107872716-70c01980-6eac-11eb-8401-738229635092.png)

short string is selected, expanding combobox
![image](https://user-images.githubusercontent.com/1220815/107872728-8af9f780-6eac-11eb-801f-4f6e9a09e549.png)

(also interesting that the combobox is cut off by the window bounds, but that's beside the point and probably falls under 'design choice' :))

**Comment by emilk** [comment]
There is no way for egui to know what the widest choice is because it doesn't run the dropdown code unless the dropdown is open. The default combo_box width is the badly named `Spacing::slider_width`, so the solution is to call `ui.spacing_mut().slider_width = 500.0;` before showing the combo box. Not extremely elegant, but it works :)

**Reply by emilk** [reply]
In the next release of egui you will be able to give an initial width with `ComboBox::from_label("Pick a card").width(200.0)…`.

Added in https://github.com/emilk/egui/commit/41c9de27341f5eec0e996390c7cf02e22949108f
